### DISCUSSION TITLE
Status of any ESTree harmonization

### DISCUSSION THREAD

**Discussion by brettz9** [babel/babel#14445] [Q&A] [answered]
Per the [Babel Roadmap](https://babeljs.io/docs/en/roadmap):

> We are trying to align our TypeScript AST with the [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint/) project. Our ASTs are almost identical, but we need to introduce some small breaking changes to fully align.

Per [@babel/parser docs](https://babeljs.io/docs/en/babel-parser#output):

> The Babel parser generates AST according to [Babel AST format](https://github.com/babel/babel/tree/main/packages/babel-parser/ast/spec.md). It is based on [ESTree spec](https://github.com/estree/estree) with the following deviations

By aligning the TypeScript AST, does this include efforts to align the other deviations from ESTree AST? If so, is there a general issue to follow to track the status? If not, might some justification be offered as to why not?

As a bonus, I'd also be curious if there were any efforts to standardize APIs for transforming, generating, etc. It seems the options at least differ across parser (e.g., `comment` vs. `comments`) and generator.

*Edit*: Such an alignment could really be a boon for ESLint I believe as the greater availability of Babel manipulation should allow ESLint rules to express themselves with such easier to follow operations as opposed to having to deal with strings and regular expressions in providing fixers.

**Comment by JLHwung** [comment, marked as answer]
> aligning the TypeScript AST

We have ongoing efforts to align the Babel TS AST to [TypeScript-ESTree](https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/ast-spec) AST, which is derived from ESTree project.

> Does this include efforts to align the other deviations from ESTree AST.

No. We don't plan to merge back to ESTree AST. Some deviations between Babel AST and ESTree AST have been there for long, such as the popular ES2022 features like class properties. Aligning these deviations will break ecosystem tremendously as every downstream Babel plugin working on class properties have to adapt to new AST.

That said, we are working with the ESTree team on early proposals so hopefully deviations between Babel AST and ESTree AST on ongoing features will be minimal, if not none.

**Reply by zm-cttae** [reply]
@JLHwung What is the status of these efforts?
